addons/project_timesheet_holidays/models/hr_holidays.py

```diff
@@ -56,52 +56,60 @@ class Holidays(models.Model):
-
-        # Unlink previous timesheets do avoid doublon (shouldn't happen on the interface but meh)
-        old_timesheets = holidays.sudo().timesheet_ids
-        if old_timesheets:
-            old_timesheets.holiday_id = False
-            old_timesheets.unlink()
+        vals_list = []
+        leave_ids = []
+        for leave in self:
+            if leave.holiday_type != 'employee' or not leave.holiday_status_id.timesheet_generate:
+                continue
 
-        # create the timesheet on the vacation project
-        holidays._timesheet_create_lines()
+            if leave.holiday_status_id.company_id:
+                project, task = leave.holiday_status_id.timesheet_project_id, leave.holiday_status_id.timesheet_task_id
+            else:
+                project, task = leave.employee_id.company_id.internal_project_id, leave.employee_id.company_id.leave_timesheet_task_id
 
-        return super()._validate_leave_request()
+            if not project or not task:
+                continue
 
-    def _timesheet_create_lines(self):
-        vals_list = []
-        for leave in self:
+            leave_ids.append(leave.id)
             if not leave.employee_id:
                 continue
+
             work_hours_data = leave.employee_id.list_work_time_per_day(
                 leave.date_from,
                 leave.date_to)
+
             for index, (day_date, work_hours_count) in enumerate(work_hours_data):
-                vals_list.append(leave._timesheet_prepare_line_values(index, work_hours_data, day_date, work_hours_count))
-        return self.env['account.analytic.line'].sudo().create(vals_list)
+                vals_list.append(leave._timesheet_prepare_line_values(index, work_hours_data, day_date, work_hours_count, project, task))
+
+        # Unlink previous timesheets to avoid doublon (shouldn't happen on the interface but meh)
+        old_timesheets = self.env["account.analytic.line"].sudo().search([('project_id', '!=', False), ('holiday_id', 'in', leave_ids)])
+        if old_timesheets:
+            old_timesheets.holiday_id = False
+            old_timesheets.unlink()
+
+        self.env['account.analytic.line'].sudo().create(vals_list)
+
+        return super()._validate_leave_request()
 
-    def _timesheet_prepare_line_values(self, index, work_hours_data, day_date, work_hours_count):
+    def _timesheet_prepare_line_values(self, index, work_hours_data, day_date, work_hours_count, project, task):
         self.ensure_one()
         return {
             'name': _("Time Off (%s/%s)", index + 1, len(work_hours_data)),
-            'project_id': self.holiday_status_id.timesheet_project_id.id,
-            'task_id': self.holiday_status_id.timesheet_task_id.id,
+            'project_id': project.id,
+            'task_id': task.id,
             'account_id': self.holiday_status_id.timesheet_project_id.analytic_account_id.id,
```

Comment:
You did the changes suggested... :/
```suggestion
            'account_id': project.sudo().analytic_account_id.id,
```

Reply:
i haven't pushed sorry :/